logicalplan/plan_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) The Thanos Community Authors.
+// Licensed under the Apache License 2.0.
+
+package logicalplan
+
+import (
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/efficientgo/core/testutil"
+	"github.com/prometheus/prometheus/promql/parser"
+)
+
+func TestDefaultOptimizers(t *testing.T) {
+	cases := []struct {
+		name     string
+		expr     string
+		expected string
+	}{
+		{
+			name:     "common selectors",
+			expr:     `sum(metric{a="b", c="d"}) / sum(metric{a="b"})`,
+			expected: `sum(filter([a="b" c="d"], metric{a="b"})) / sum(metric{a="b"})`,
```

Comment:
Do we need to keep the old matchers in the filter when the vector selector has them? For example, I'd expect to see

```go
			expected: `sum(filter([c="d"], metric{a="b"})) / sum(metric{a="b"})`,
```

Here? :thinking:

Reply:
That's a good point - I guess we do double selection here which is not a harm - we can optimize this later (:

Reply:
True, it would be safe and better to strip selectors from filters. It will make also make filtering faster.

Reply:
This ended up being fairly straightforward, so I implemented it in this PR.